src/network/networkd-dhcp4.c

```diff
@@ -138,6 +139,71 @@ static bool link_prefixroute(Link *link) {
                 link->manager->dhcp4_prefix_root_cannot_set_table;
 }
 
+static int dhcp_set_routing_policy_rules(Link *link) {
+        const struct in_addr *router = NULL;
+        struct in_addr address, netmask;
+        RoutingPolicyRule *rule;
+        unsigned prefixlen;
+        int r;
+
+        assert(link);
+
+        r = sd_dhcp_lease_get_address(link->dhcp_lease, &address);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "DHCP error: no address: %m");
+
+        r = sd_dhcp_lease_get_netmask(link->dhcp_lease, &netmask);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "DHCP error: no netmask: %m");
+
+        prefixlen = in4_addr_netmask_to_prefixlen(&netmask);
+
+        (void) sd_dhcp_lease_get_router(link->dhcp_lease, &router);
+
+        if (DEBUG_LOGGING) {
+                _cleanup_free_ char *pretty = NULL;
+                union in_addr_union u = {
+                        .in = address,
+                };
+
+                (void) in_addr_to_string(AF_INET, &u, &pretty);
+                log_link_debug(link, "Configuring RPDB rule from DHCPv4 address %s", strna(pretty));
+        }
+
+        HASHMAP_FOREACH(rule, link->network->rules_by_section) {
+                if (!rule->from_dhcp_ip && !rule->to_dhcp_ip && !rule->from_dhcp_gateway && !rule->to_dhcp_gateway)
+                        continue;
+
+                rule->family = AF_INET;
+
```

Comment:
Please refuse the case that `from_dhcp_gateway` or `to_dhcp_gateway` is enabled bu the lease does not have router:
```
if ((rule->from_dhcp_gateway || rule->to_dhcp_gateway) && !router) {
        log_link_debug(link, "...");
        continue;
}
```